mobile/lib/features/welcome/welcome_screen.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:get_10101/util/preferences.dart';
+import 'package:go_router/go_router.dart';
+import 'package:get_10101/ffi.dart';
+
+class WelcomeScreen extends StatefulWidget {
+  static const route = "/welcome";
+  static const label = "Welcome";
+
+  const WelcomeScreen({Key? key}) : super(key: key);
+
+  @override
+  State<WelcomeScreen> createState() => _WelcomeScreenState();
+}
+
+class _WelcomeScreenState extends State<WelcomeScreen> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+
+  String _email = "";
+
+  /// TODO Convert to a flutter package that checks the email domain validity
+  /// (MX record, etc.)
+  bool isEmailValid(String email) {
+    return RegExp(r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]+")
+        .hasMatch(email);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Form(
+      key: _formKey,
+      child: Column(
+        children: <Widget>[
+          Column(
+            children: const <Widget>[
+              Text(
+                'Welcome to 10101 closed beta!',
+                style: TextStyle(
+                  fontSize: 20,
+                  fontWeight: FontWeight.bold,
+                ),
+              ),
+              SizedBox(height: 10),
+              Text("Please be patient with us as we work out the rough edges."),
+              Text("Any feedback is welcome!"),
+              SizedBox(height: 10),
+              Text("Please enter your email address to continue:"),
+            ],
+          ),
+          const SizedBox(height: 20),
+          TextFormField(
+            keyboardType: TextInputType.emailAddress,
+            initialValue: _email,
+            decoration: const InputDecoration(
+              labelText: 'Email',
+              hintText: 'Enter your email address',
+            ),
+            validator: (value) {
+              if (value == null || value.isEmpty || !isEmailValid(value)) {
+                return 'Please enter a valid email address';
+              }
+              return null;
+            },
+            onSaved: (value) {
+              _email = value ?? "";
+            },
+          ),
+          ElevatedButton(
+            onPressed: () {
+              if (_formKey.currentState != null && _formKey.currentState!.validate()) {
+                _formKey.currentState?.save();
+                Preferences.instance.setEmailAddress(_email);
+                FLog.info(text: "Successfully stored the email address $_email .");
+                api.registerBeta(email: _email);
+                context.go(WalletScreen.route);
+              }
+            },
+            child: const Text('Start'),
+          ),
+        ],
+      ),
+    );
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    Preferences.instance.getEmailAddress().then((value) => setState(() {
+          FLog.info(text: "retrieved stored email from the preferences: $_email.");
```

Comment:
I think you want to print `value` here as `_email` is only set after the log message. 
Or move the log message down.

Reply:
> I think you want to print `value` here as `_email` is only set after the log message.
> Or move the log message down.

omg, thank you. nothing was wrong, I just read the wrong field. 🙈